Spigot-API-Patches/0038-Add-proper-Entity-Player-OfflinePlayer-equals-and-ha.patch

```diff
@@ -0,0 +1,191 @@
+From b56507323afd7a004be1b872141a7eef0f48909a Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Searchndstroy@users.noreply.github.com>
+Date: Sun, 24 Apr 2016 01:23:11 -0700
+Subject: [PATCH] Add proper Entity/Player/OfflinePlayer equals and hashCode
+ specifications to Paper API
+
+Entity hashCode specifies that the implementation should implement the hash
+code only including the entity id (see Javadoc for the specifics).
+
+Entity equals specifies that the implementation should implemnet the equality
+algorithm by checking 'instanceof Entity' and entity id (see Javadoc).
+
+The Player interface uses the hashCode specification from OfflinePlayer to
+retain compatibility. To ensure that the Object.equals specification is not
+broken as well as possible backwards compatibility, the Player.equals specification
+aligns with the previous implementation of Player.equals.
+
+The OfflinePlayer interface equals and hashCode specification align with the
+(un-changed) direct OfflinePlayer implementation.
+
+diff --git a/src/main/java/org/bukkit/OfflinePlayer.java b/src/main/java/org/bukkit/OfflinePlayer.java
+index e98706a..f17c684 100644
+--- a/src/main/java/org/bukkit/OfflinePlayer.java
++++ b/src/main/java/org/bukkit/OfflinePlayer.java
+@@ -115,4 +115,46 @@ public interface OfflinePlayer extends ServerOperator, AnimalTamer, Configuratio
+      */
+     public Location getBedSpawnLocation();
+ 
++    // Paper start - define equals and hashCode for special case with Entity
++
```

Comment:
I've fixed the extra lines.

Consider clearly documenting "your [Paper's] style" if it cares about pendant details such as inserting new lines after "// Paper start"

The CONTRIBUTING.md does **not** make such a documentation, nor does it refer to any type of document, article or piece of information which does.